### PR TITLE
Cleanup and reorganize all the helm related code

### PR DESCRIPTION
**Pull request by rohitkrai03**
### This PR - 
- Just cleans up the code and re-organize all the files related to Helm. No visible changes are there. 
- Just a bunch of moving and renaming of components, util functions and files into an organized structure.

#### Before
Screenshot from 2020-04-03 21-26-58

#### After
Screenshot from 2020-04-03 21-26-09

### CODE REVIEW

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx [hunk 1]
```diff
@@ -1,144 +1,52 @@
...
-import { deleteHelmRelease, upgradeHelmRelease } from '../../actions/modify-helm-release';
-import HelmReleaseNotes from './HelmReleaseNotes';
-import { HelmRelease } from './helm-types';
+import HelmReleaseDetails from './details/HelmReleaseDetails';
 
-const SecretReference: K8sResourceKindReference = 'Secret';
-const HelmReleaseReference = 'HelmRelease';
-export interface HelmReleaseDetailsPageProps {
+export const HELMRELEASES_ALL_NS_PAGE_URI = '/helm-releases/all-namespaces';
+export interface HelmReleaseDetailsProps {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx:10]: nit : we may not need `export` here

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx [hunk 2]
```diff
@@ -1,144 +1,52 @@
...
+        variant={NamespacedPageVariants.light}
+        hideApplications
+        onNamespaceChange={handleNamespaceChange}
+      >
+        <Firehose
+          resources={[
+            {
+              namespace,
+              kind: SecretModel.kind,
+              prop: 'secret',
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx:40]: can use `SecretModel.id` here

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx [hunk 3]
```diff
@@ -1,144 +1,52 @@
...
-    getHelmReleases();
-
-    return () => {
-      ignore = true;
-    };
-  }, [helmReleaseName, namespace]);
-
   return (
-    <LoadedHelmReleaseDetailsPage match={match} secret={secret} helmReleaseData={helmReleaseData} />
+    <>
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx:29]: nit: can avoid empty fragment

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx (hunk 1 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx:10]: and can we call it `HelmReleaseDetailsPageProps` as we have for other component any reason for having it as above

File: frontend/packages/dev-console/src/components/helm/__tests__/helm-release-mock-data.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { HelmRelease, HelmChartMetaData } from '../helm-types';
+import { K8sResourceKind } from '@console/internal/module/k8s';
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/__tests__/helm-release-mock-data.ts:2]: nit : can have above relative imports

File: frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+import HelmReleaseResources from './resources/HelmReleaseResources';
+import HelmReleaseOverview from './overview/HelmReleaseOverview';
+import HelmReleaseHistory from './history/HelmReleaseHistory';
+import { deleteHelmRelease, upgradeHelmRelease } from '../../../actions/modify-helm-release';
+import HelmReleaseNotes from './HelmReleaseNotes';
+import { HelmRelease } from '../helm-types';
+
+const SecretReference: K8sResourceKindReference = 'Secret';
+const HelmReleaseReference = 'HelmRelease';
+export interface HelmReleaseDetailsProps {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx:26]: nit: can avoid `export`

File: frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  match: RMatch<{
+    ns?: string;
+    name?: string;
+  }>;
+  secret?: FirehoseResult;
+}
+
+export interface LoadedHelmReleaseDetailsProps extends HelmReleaseDetailsProps {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx:34]: nit: can avoid export and will it make sense to have `LoadedHelmReleaseDetails` in diff file and have `HelmReleaseDetails` here

File: frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+      customKind={HelmReleaseReference}
+    />
+  );
+};
+
+const HelmReleaseDetails: React.FC<HelmReleaseDetailsProps> = ({ secret, match }) => {
+  const namespace = match.params.ns;
+  const helmReleaseName = match.params.name;
+
+  const [helmReleaseData, setHelmReleaseData] = React.useState<any>();
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx:117]: can we have type here

File: frontend/packages/dev-console/src/components/helm/details/resources/__tests__/HelmResourcesList.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Table } from '@console/internal/components/factory';
+import HelmResourcesList from '../HelmReleaseResourcesList';
+import HelmReleaseResourcesHeader from '../HelmReleaseResourcesHeader';
+import HelmReleaseResourcesRow from '../HelmReleaseResourcesRow';
+
+describe('HelmResourcesListComponent', () => {
+  const helmResourcesList = shallow(
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/details/resources/__tests__/HelmResourcesList.spec.tsx:9]: nit : we can have it in  `beforeEach` or  `beforeAll`

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx [hunk 9]
```diff
@@ -1,144 +1,52 @@
...
 import { match as RMatch } from 'react-router';
-import {
-  navFactory,
-  LoadingBox,
-  StatusBox,
-  FirehoseResult,
-} from '@console/internal/components/utils';
+import { Firehose, history } from '@console/internal/components/utils';
+import { ALL_NAMESPACES_KEY } from '@console/shared';
+import NamespacedPage, { NamespacedPageVariants } from '../NamespacedPage';
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx:5]: nit: can move it below

File: frontend/packages/dev-console/src/components/helm/__tests__/HelmReleaseDetailsPage.spec.tsx [hunk 10]
```diff
@@ -1,82 +1,27 @@
...
-                owner: 'helm',
-                status: 'deployed',
-              },
-            },
-          },
-        ],
-        loadError: null,
-        loaded: true,
+describe('HelmReleaseDetails', () => {
+  helmReleaseDetailsProps = {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/__tests__/HelmReleaseDetailsPage.spec.tsx:10]: nit: can have it in `beforeEach`

File: frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx (hunk 6 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/helm/details/HelmReleaseDetails.tsx:34]: The components are too tightly coupled to have it in diff files IMO.

File: frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx (hunk 2 above)
Comment (christianvogt) [reply frontend/packages/dev-console/src/components/helm/HelmReleaseDetailsPage.tsx]: To be fair, this is very specific to the individual use of the result data.
Using an arbitrary string is perfectly fine here.

File: frontend/packages/dev-console/src/components/helm/__tests__/HelmReleaseDetailsPage.spec.tsx [hunk 11]
```diff
@@ -1,82 +1,32 @@
...
-import HelmReleaseDetailsPage, { LoadedHelmReleaseDetailsPage } from '../HelmReleaseDetailsPage';
-import { mockHelmReleases } from './helm-release-mock-data';
+import NamespacedPage from '../../NamespacedPage';
+import HelmReleaseDetails from '../details/HelmReleaseDetails';
+import HelmReleaseDetailsPage from '../HelmReleaseDetailsPage';
 
-let helmReleaseDetailsPageProps: React.ComponentProps<typeof HelmReleaseDetailsPage>;
-let loadedHelmReleaseDetailsPageProps: React.ComponentProps<typeof LoadedHelmReleaseDetailsPage>;
+let helmReleaseDetailsProps: React.ComponentProps<typeof HelmReleaseDetailsPage>;
+let helmReleaseDetailsPage;
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/__tests__/HelmReleaseDetailsPage.spec.tsx:8]: nit : can have type, `let helmReleaseDetailsPageWrapper = ShallowWrapper< helmReleaseDetailsProps >;`

File: frontend/packages/dev-console/src/components/helm/details/resources/__tests__/HelmResourcesList.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Table } from '@console/internal/components/factory';
+import HelmResourcesList from '../HelmReleaseResourcesList';
+import HelmReleaseResourcesHeader from '../HelmReleaseResourcesHeader';
+import HelmReleaseResourcesRow from '../HelmReleaseResourcesRow';
+
+describe('HelmResourcesList', () => {
+  let helmResourcesList;
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/details/resources/__tests__/HelmResourcesList.spec.tsx:9]: nit: type for `helmResourcesList`

**Comment by invincibleJai** [comment]
Verified locally working as expected

**Pull request merged**